docs/source/dev-docs.rst

```diff
@@ -127,6 +127,95 @@ to run each do the following:
+
+:code:`AppTestBase` contains for members accessible to your tests
+
+
++--------------------------------------------+--------------------------------+
+|Member Variable                             |Description                     |
++============================================+================================+
+|@SystemStub\                                |variables from \                |
+|protected final EnvironmentVariables \      |System.getenv \                 |
+|environment = new EnvironmentVariables();   |that applications will see.     |
++--------------------------------------------+--------------------------------+
+|@SystemStub\                                |variables from \                |
+|protected final SystemProperties \          |System.getProperty \            |
+|properties = new SystemProperties();        |that applications will see.     |
++--------------------------------------------+--------------------------------+
+|@SystemStub\                                |Used to trap System.exit        |
+|protected final SystemExit \                |calls to allow testing          |
+|exit = new SystemExit();                    |without aborting the test run   |
++--------------------------------------------+--------------------------------+
+|@ConfiguredField                            |Instance of the                 |
+|protected Configuration configuration;      |:code:`Configuration` that was  |
+|                                            |create for this run. Contains   |
+|                                            |reference to user.properties and|
+|                                            | other specific information     |
++--------------------------------------------+--------------------------------+
+ 
+Extension and other Junit information
+-------------------------------------
+
+The :code:`OpenDCSTestConfigExtension`, if it knows about a given type, will inject 
+an instance of any field annotated with :code:`@ConfiguredField` as seen in the table 
+above for the configuration.
+
+The only other injected field is a :code:`TimeSeriesDb` which is Provided by the Configuration
+and will already be valid and can be used directly for things like testing DaoObjects or null which 
+indicates the implementation under test doesn't use the any of the timeseries database components.
+
+The sample :code:`LoadingAppDaoTestIT` uses the :code:`@EnableIfSql` :code:`ExecutionCondition`
+ to determine if the test should be run or not.
+
+Additional ExecutionConditions and parameter injection will be added in the future as needed and as
+we better identify concepts to map to vs implementation details.
+
+
+Caveats
+-------
+
+OpenDCS supports several implementations, the XML database, the baseline Postgres and Oracle database,
+USBR's HDB and USACE's CWMS. While each share the same fundamental concepts portions of the implementation, like Site names
```

Comment:
< line feed >


Each implementation shares the same fundamental concepts.  However, portions of the